src/parquet/column/levels.h

```diff
@@ -95,23 +95,32 @@ class LevelDecoder {
  public:
   LevelDecoder() {}
 
-  // Initialize the LevelDecoder and return the number of bytes consumed
-  size_t Init(parquet::Encoding::type encoding, int16_t max_level,
+  // Initialize the LevelDecoder
+  void Init(int16_t max_level) {
+    bit_width_ = BitUtil::Log2(max_level + 1);
+    rle_decoder_.reset(new RleDecoder(NULL, 0, 0));
+    bit_packed_decoder_.reset(new BitReader(NULL, 0));
```

Comment:
I couldn't find any details about the encoding switch.
However, the issue here is that the level encoding is specified in the `DataPageHeader`. This is probably to allow different encodings mid-stream. The `Init` is called once during the instantiation of the column reader. I will have a check for `max_level > 0` there.